scripts/export-openapi-cli.js

```diff
@@ -0,0 +1,349 @@
+           exampleValue == undefined anyway
+        */
+        const pathParams = []
+        for (const param of openApiPattern
+          .split('/')
+          .filter(p => p.startsWith('{') && p.endsWith('}'))) {
+          const paramName = param.slice(1, -1)
+          const exampleValue = example.example.namedParams[paramName]
+          pathParams.push(
+            param2openapi(pattern, paramName, exampleValue, 'path')
+          )
+        }
+
+        const queryParams = example.example.queryParams || {}
+
+        const parameters = [
+          ...pathParams,
+          ...Object.entries(queryParams).map(([paramName, exampleValue]) =>
+            param2openapi(pattern, paramName, exampleValue, 'query')
+          ),
+          { $ref: '#/components/parameters/style' },
+          { $ref: '#/components/parameters/logo' },
+          { $ref: '#/components/parameters/logoColor' },
+          { $ref: '#/components/parameters/label' },
+          { $ref: '#/components/parameters/labelColor' },
+          { $ref: '#/components/parameters/color' },
+          { $ref: '#/components/parameters/cacheSeconds' },
+          { $ref: '#/components/parameters/link' },
+        ]
+        paths[openApiPattern] = {
+          get: {
+            summary: example.title,
+            description: example?.documentation?.__html
+              .replace(/<br>/g, '<br />') // react does not like <br>
+              .replace(/{/g, '&#123;')
+              .replace(/}/g, '&#125;')
+              .replace(/<style>(.|\n)*?<\/style>/, ''), // workaround for w3c-validation TODO: remove later
+            parameters,
+            'x-code-samples': getCodeSamples(example.title),
+          },
+        }
+      }
+    }
+  }
+  return paths
+}
+
+function category2openapi(category, services) {
+  const spec = {
+    openapi: '3.0.0',
+    info: {
+      version: '1.0.0',
+      title: category.name,
+      license: {
+        name: 'CC0',
+      },
+    },
+    servers: [{ url: baseUrl }],
+    components: {
+      parameters: {
+        style: {
+          name: 'style',
+          in: 'query',
+          required: false,
+          description:
+            'One of: flat (default), flat-square, plastic, for-the-badge, social',
+          schema: {
+            type: 'string',
+          },
+          example: 'flat',
+        },
+        logo: {
+          name: 'logo',
+          in: 'query',
+          required: false,
+          description:
+            'Insert one of the named logos (bitcoin, dependabot, gitlab, npm, paypal, serverfault, stackexchange, superuser, telegram, travis) or <a href="https://simpleicons.org/" rel="noopener noreferrer" target="_blank">simple-icons</a>. Simple-icons are referenced using icon slugs which can be found on the simple-icons site or in the <a href="https://github.com/simple-icons/simple-icons/blob/develop/slugs.md" rel="noopener noreferrer" target="_blank">slugs.md</a> file in the simple-icons repository.',
```

Comment:
Perhaps we should go ahead and change this to use the SI master branch too given #9054

```suggestion
            'Insert one of the named logos (bitcoin, dependabot, gitlab, npm, paypal, serverfault, stackexchange, superuser, telegram, travis) or <a href="https://simpleicons.org/" rel="noopener noreferrer" target="_blank">simple-icons</a>. Simple-icons are referenced using icon slugs which can be found on the simple-icons site or in the <a href="https://github.com/simple-icons/simple-icons/blob/master/slugs.md" rel="noopener noreferrer" target="_blank">slugs.md</a> file in the simple-icons repository.',
```

Reply:
Ah the bane of commit-by-commit reviews :laughing: